### PR TITLE
DAOS-9365 test: pool/permission.py - skipped variants should be removed.

### PR DESCRIPTION
**Pull request by dinghwah**
Description: 
Removed cancel_for_ticket after DAOS-3442 resolved.
Removed redundant codes.
Pool test_connect is a subset of  test_file_modification, merged the 2 tests into 1.

Quick-Functional: true
Test-tag: permission
Test-repeat: 5
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

### CODE REVIEW

**Review by rpadma2 — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/pool/permission.yaml [hunk 1]
```diff
@@ -19,8 +19,6 @@ pool:
     name: daos_server
 createtests:
     createmode: !mux
-        mode_RO:
-            mode: 73
```
Comment (mjean308) [inline src/tests/ftest/pool/permission.yaml:23]: Is 73 a valid mode?   If not should the test code conditional be removed all together?
``        if createmode == 73:
            permissions = 0
            expected_result = RESULT_FAIL``

Comment (dinghwah) [reply src/tests/ftest/pool/permission.yaml:23]: please see the update.

**Review by shimizukko — DISMISSED** [review]

**Review by phender — CHANGES_REQUESTED** [review]
Why are we removing the negative test case from this test?  Doesn't the `createmode == 73` verify that if an invalid mode is provided that the pool connect fails as expected?

==>This per request DAOS-9365.

File: src/tests/ftest/pool/permission.yaml (hunk 1 above)
Comment (mjean308) [reply src/tests/ftest/pool/permission.yaml:23]: I see it is expected to fail...

**Review by mjean308 — APPROVED** [review]

**Review by phender — CHANGES_REQUESTED** [review]
After looking at this in more detail, this test requires more changes.   The *entire* `createmode` mode should be removed as it has no use when running `dmg pool create` - it was used when we used to have an API pool create method, but that no longer exists.

With the `createmode` code and yaml entries all removed, what is expected to happen with the `self.pool.connect(1 << permissions)` command?  Are there any cases where we expect this to pass or fail?  Do we need  to pass additional options (`--group`, `--user`,  `--permissions`, or `--acl_file`?) to the `dmg pool create` command to be able to issue positive and negative test cases?

==>It is correct, I think this old test is not targeting on pool create with createmode, so there is no pass/fail check.
       After pool created, this test is testing pool connect with different permission, without acl, user/group, which is covered in security.

**Review by shimizukko — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Review by phender — CHANGES_REQUESTED** [review]
Something odd is going on with this test.  In https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-7915/6/artifact/Functional%20on%20CentOS%208/pool/permission.py/1/job.log the `1-./pool/permission.py:Permission.test_connect_permission`, for example,  expects the pool connect to fail:
```
2022-02-08 18:21:11,468 parameters       L0143 DEBUG| PARAMS (key=perm, path=/run/createtests/permissions/*, default=None) => 0
2022-02-08 18:21:11,468 parameters       L0143 DEBUG| PARAMS (key=exp_result, path=/run/createtests/permissions/*, default=None) => 'FAIL'
```
The pool connect passes and the test identifies the situation correctly as a test failure:
```
2022-02-08 18:21:14,134 test_utils_pool  L0293 INFO | Connecting to pool 99C65748-7084-4E2B-9034-0BE2B85FBF16 with permission 2 (flag: 2)
2022-02-08 18:21:14,376 logger_utils     L0037 DEBUG| Pool Creation successful
2022-02-08 18:21:14,378 logger_utils     L0037 DEBUG| Pool Connect successful
2022-02-08 18:21:14,379 permission       L0054 ERROR| Test was expected to fail at pool.connect, but it passed.
```
however, the test is reported as a PASS:
```
2022-02-08 18:21:16,269 test             L0939 INFO | PASS 1-./pool/permission.py:Permission.test_connect_permission;run-createtests-permissions-perm_RO-hosts-pool-server_config-635e
```

==>Thanks for the detail review and good catch for the error message.
==>Actually, the 1st permission test pool.connect() are subset of the 2nd test file_modification test, I merge 2 tests into one and also added more debug-log when test passed.

**Review by shimizukko — APPROVED** [review]

**Review by rpadma2 — APPROVED** [review]

**Pull request merged**